lib/net/message-socket.d.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2016-2018 Untu, Inc.
+ * This code is licensed under Eclipse Public License - v 1.0.
+ * The full license text can be found in LICENSE.txt file and
+ * on the Eclipse official site (https://www.eclipse.org/legal/epl-v10.html).
+ */
+
+import { EventEmitter } from 'events';
+import {Socket} from 'net';
+
+export class MessageSocket extends EventEmitter {
+
+  constructor(socket: Socket);
+  // @ts-ignore
+  on(event: string | symbol, listener: (...args: any[]) => void): void;
+  // @ts-ignore
+  once(event: string | symbol, listener: (...args: any[]) => void): void;
+  // @ts-ignore
+  removeListener(event: string | symbol, listener: (...args: any[]) => void): void;
+
+  write(message: object, cb: Function): void;
+  send(message: object, cb?: Function): void;
+  makePacket(message: object): Buffer;
+  end(): void;
+  destroy(): void;
+  _handleIncomingData(data: Buffer): void;
```

Comment:
Are we using this method publicly? If not, it should be marked as `private` or completely removed from type definition (the latter makes more sense, I guess).

Reply:
Seems we do use `makePacket()` in tests, but not elsewhere. Private won't work in this case. I've deleted this method from the definition but now thinking about restoring it.

Reply:
I was talking about `_handleIncomingData()`. Please note that the comment also refers to a single line above.

Reply:
Also note that we only need TypeScript tests for publicly available stuff. As MessageSocket is a completely internal class not seen by client, it's tests can be safely left in JavaScript.